mne/viz/topomap.py

```diff
@@ -41,8 +41,11 @@ def _prepare_topo_plot(inst, ch_type, layout):
         layout = find_layout(info)  # XXX : why not passing ch_type???
     elif layout == 'auto':
         layout = None
-
-    clean_ch_names = _clean_names(info['ch_names'])
+    if ch_type in ['hbo', 'hbr', 'fnirs_raw', 'fnirs_od']:
+        # The naming for nirs is very specific and should not be modified
+        clean_ch_names = info['ch_names']
+    else:
+        clean_ch_names = _clean_names(info['ch_names'])
```

Comment:
_clean_names is used in a number of places. I would rather avoid the branching here to make sure it's consistent with other places. How specific are NIRS channel names? What ch name breaks it?

Reply:
The name given by the device manufacturer is in the format of source number then detector number, so for example “S1 D2” is the channel between source one and detector two. We then appended the light frequency or chomaphore. So currently the MNE channel names are in the format `S1 D2 760`.

The problem was that the spaces were getting removed and the first characters were being kept, so we would have several channel names that were modified to be “S1” for example. And that caused the error listed in the first comment of this PR.

But we can change the way channel names are stored. We could make it `S1-D2-760` Or `S1D2-760` etc. I think this would solve the problem, but not sure if it will break other things. What do you think?

Reply:
Sorry that’s wrong, the format is currently `S1-D2 760`. Another example is `S12-D3 859`

https://github.com/mne-tools/mne-python/blob/master/mne/io/nirx/tests/test_nirx.py#L37

Reply:
Ok I have push an attempt to change the naming to `S1_D3 760`. This way the source-detector pair is read as a single word, so stays unique without modifying clean_names. The raw.plot doesnt look as good, but I think thats ok. Thoughts?